compass/landice/tests/dome/run_model.py

```diff
@@ -72,6 +77,11 @@ def __init__(self, test_case, mesh_type, name='run_model', subdir=None,
                 'compass.landice.tests.dome', 'namelist.landice',
                 out_name='namelist.{}'.format(suffix))
 
+            replacements = {'config_velocity_solver': '{}'.format(velo_solver)}
+
+#            update(replacements=replacements, step_work_dir=self.path,
```

Comment:
@xylar , when I uncomment this line, `compass list` fails with this error:
```
FileNotFoundError: [Errno 2] No such file or directory: 'landice/dome/sia/2000m/smoke_test/run_model/namelist.landice'
```
My guess is that the namelist `update` method doesn't work in this context because self.add_namelist_file hasn't actually done anything yet in `compass list`.  I may need to rethink how this is organized.  I don't want to wait to update the namelist until the `run` method, because then the namelist set up in the working directory would be wrong and likely confusing if anyone looked at it.  What do you suggest?  Is it actually necessary for `compass list` to run a step's `__init__` function?

Reply:
I can understand the confusion.  There is a method `self.add_namelist_options()` that is what you want here.  

https://mpas-dev.github.io/compass/latest/developers_guide/organization.html#adding-namelist-options

`compass.namelist.update()` only works in `run()` as you have discovered.  It would be really helpful if we could come up with a way to make these two more consistent.  One thing would be to move `update()` to be a method in `Step` as well, maybe named something like `update_namelist_at_runtime()`.  Verbose but hopefully clearer.

I'd appreciate your thoughts on what would clear this up for you.

Reply:
I like the idea of renaming it `update_namelist_at_runtime()`.  And if the function can really only be applied to a step then moving into the `Step` class where it would exist alongside `add_namelist_options()` would be clean.

By the way, I did have a look in the docs to try to answer my question but for some reason didn't find the section that holds the answer.  I found this section instead and stopped looking: https://mpas-dev.github.io/compass/latest/developers_guide/framework.html#updating-a-namelist-file  I just noticed now the docs have a search function, so I should remember that in the future.

Reply:
I have a fix for this mostly implemented.  I don't think this PR would be affected by it, since you're now using `add_namelist_options()`, right?

Reply:
That is correct.  I will proceed to merge without that change.